modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateProcessor.java

```diff
@@ -121,7 +122,16 @@ public DateProcessor create(Map<String, Processor.Factory> registry, String proc
             Locale locale = Locale.ENGLISH;
             if (localeString != null) {
                 try {
-                    locale = (new Locale.Builder()).setLanguageTag(localeString).build();
```

Comment:
would it work to use `LocalUtils.parse(localeString)` like we do for the date field?